### PR TITLE
[C-1672] TikTok verification via Settings page

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Add TikTok verification to the verification flow from settings page on web and mobile web
* Add TikTok verification to the SocialProof flow from the Audio page
* Refactor social auth buttons to share styles
  * `SocialButton` encapsulates styles
  * `TwitterButton`, `InstagramButton`, `TikTokButton` are dumb components that specify a `SocialButton`
  * `TwitterAuthButton` etc. are smart components that actually trigger the auth via the corresponding button

### Dragons

Touches all the auth flows, specifically changes Twitter auth to call onSuccess with a `uuid` and a `profile` like the other two do. Everything seems to work but will want to test thoroughly before release

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/components/social-button/SocialButton.module.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  height: 20px !important;
+  width: 20px !important;
+}
+
+.socialButton .icon path {
+  fill: var(--static-white);
+}
+
+.text {
+  font-size: var(--font-xl) !important;
```
Comment (dylanjeffers) [inline packages/web/src/components/social-button/SocialButton.module.css:17]: one day we can address non needing !important :/ i guess it has to do with our button style specificity?

File: packages/web/src/components/tiktok-auth/TikTokAuthButton.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const TikTokAuthButton = (props: TikTokAuthButtonProps) => {
+  const { onFailure, onSuccess, onClick, ...buttonProps } = props
+
+  const withTikTokAuth = useTikTokAuth({
+    onError: onFailure
+  })
+
+  const handleClick = useCallback(
+    (e: any) => {
```
Comment (dylanjeffers) [inline packages/web/src/components/tiktok-auth/TikTokAuthButton.tsx:24]: MouseEvent maybe?

File: packages/web/src/components/tiktok-auth/TikTokAuthButton.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      onClick?.(e)
+      withTikTokAuth(async (accessToken: string) => {
```
Comment (dylanjeffers) [inline packages/web/src/components/tiktok-auth/TikTokAuthButton.tsx:26]: the tiktok auth fn is wild! good stuff

File: packages/web/src/pages/settings-page/components/desktop/VerificationModal.tsx [hunk 4]
```diff
@@ -256,6 +292,29 @@ const VerificationModal = (props: VerificationModalProps) => {
     [dispatch, handle, onTwitterLogin, setError, record]
   )
 
+  const tikTokLogin = useCallback(
```
Comment (dylanjeffers) [inline packages/web/src/pages/settings-page/components/desktop/VerificationModal.tsx:295]: nice

File: packages/web/src/pages/sign-on/components/CompleteProfileWithSocial.tsx [hunk 5]
```diff
@@ -70,50 +72,20 @@ const CompleteProfileWithSocial = (props: CompleteProfileWithSocialProps) => {
...
 
-  const onClickInstagram = () => {
+  const handleClickInstagram = useCallback(() => {
     onInstagramStart()
     onClick()
-  }
+  }, [onInstagramStart, onClick])
 
   const handleClickTikTok = useCallback(() => {
+    onTikTokStart()
```
Comment (dylanjeffers) [inline packages/web/src/pages/sign-on/components/CompleteProfileWithSocial.tsx:86]: nice cleanup

**Review by dylanjeffers — COMMENTED** [review]
very clean, nice work

File: packages/web/src/components/social-button/SocialButton.module.css (hunk 1 above)
Comment (sliptype) [reply packages/web/src/components/social-button/SocialButton.module.css:17]: Yeah it was either that or adding extra classes to out-specify the styles that are on the button :(

css in js wen

File: packages/web/src/components/instagram-auth/InstagramAuthButton.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+export const InstagramAuthButton = (props: InstagramAuthButtonProps) => {
+  const {
+    containerClassName,
+    onSuccess,
+    onFailure,
+    disabled = false,
+    ...buttonProps
+  } = props
+  return (
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/web/src/components/instagram-auth/InstagramAuthButton.tsx:27]: Added it but github isn't showing it for some reason, it's definitely in the code though https://github.com/AudiusProject/audius-client/pull/2488/commits/80c496159ae1be705237a9f7c292b299f27ffe3d

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-c-1672-settings-verification

**Review by dylanjeffers — APPROVED** [review]
Awesome

**Pull request merged**